instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/InstrumenterBuilder.java

```diff
@@ -190,39 +190,57 @@ public InstrumenterBuilder<REQUEST, RESPONSE> setEnabled(boolean enabled) {
    * and inject context into requests.
    */
   public Instrumenter<REQUEST, RESPONSE> buildClientInstrumenter(TextMapSetter<REQUEST> setter) {
-    return buildInstrumenter(
-        InstrumenterConstructor.propagatingToDownstream(requireNonNull(setter, "setter")),
-        SpanKindExtractor.alwaysClient());
+    return buildOngoingInstrumenter(
+        requireNonNull(setter, "setter"), SpanKindExtractor.alwaysClient());
   }
 
   /**
    * Returns a new {@link Instrumenter} which will create {@linkplain SpanKind#SERVER server} spans
    * and extract context from requests.
    */
   public Instrumenter<REQUEST, RESPONSE> buildServerInstrumenter(TextMapGetter<REQUEST> getter) {
-    return buildInstrumenter(
-        InstrumenterConstructor.propagatingFromUpstream(requireNonNull(getter, "getter")),
-        SpanKindExtractor.alwaysServer());
+    return buildIncomingInstrumenter(
+        requireNonNull(getter, "getter"), SpanKindExtractor.alwaysServer());
   }
 
   /**
    * Returns a new {@link Instrumenter} which will create {@linkplain SpanKind#PRODUCER producer}
    * spans and inject context into requests.
    */
   public Instrumenter<REQUEST, RESPONSE> buildProducerInstrumenter(TextMapSetter<REQUEST> setter) {
-    return buildInstrumenter(
-        InstrumenterConstructor.propagatingToDownstream(requireNonNull(setter, "setter")),
-        SpanKindExtractor.alwaysProducer());
+    return buildOngoingInstrumenter(
+        requireNonNull(setter, "setter"), SpanKindExtractor.alwaysProducer());
   }
 
   /**
    * Returns a new {@link Instrumenter} which will create {@linkplain SpanKind#CONSUMER consumer}
    * spans and extract context from requests.
    */
   public Instrumenter<REQUEST, RESPONSE> buildConsumerInstrumenter(TextMapGetter<REQUEST> getter) {
+    return buildIncomingInstrumenter(
+        requireNonNull(getter, "getter"), SpanKindExtractor.alwaysConsumer());
+  }
+
+  /**
+   * Returns a new {@link Instrumenter} which will create spans with kind determined by the passed
+   * {@link SpanKindExtractor} and inject context into requests.
+   */
+  public Instrumenter<REQUEST, RESPONSE> buildOngoingInstrumenter(
```

Comment:
This method is not used at this time, but I added it to make it symmetrical with the buildIncomingInstrumenter.